tests/test_exceptions/test_validation_error.py

```diff
@@ -14,3 +14,83 @@ def test_messages_concatenates_error_dict_values(self):
+
+    def test_error_dict_message_equality(self):
+        exception_a = ValidationError({'field1': 'field error', 'field2': 'err'})
+        exception_b = ValidationError({'field2': 'err', 'field1': 'field error'})
+        self.assertEqual(exception_a, exception_b)
+        self.assertEqual(exception_a.__hash__(), exception_b.__hash__())
+
+        exception_b = ValidationError({'field1': 'field error', 'field2': 'error 2'})
+        self.assertNotEqual(exception_a, exception_b)
+        self.assertNotEqual(exception_a.__hash__(), exception_b.__hash__())
+
+        exception_b = ValidationError({'field1': 'field error'})
+        self.assertNotEqual(exception_a, exception_b)
+        self.assertNotEqual(exception_a.__hash__(), exception_b.__hash__())
+
+        error_dict = {
+            'field1': ValidationError('nice error %(parm1)s', code='my_code', params={'parm1': 'val1'}),
+            'field2': 'err'
+        }
+        exception_a = ValidationError(error_dict)
+        exception_b = ValidationError(dict(error_dict))
+        self.assertEqual(exception_a, exception_b)
+        self.assertEqual(exception_a.__hash__(), exception_b.__hash__())
+
+    def test_error_list_message_equality(self):
+        error_list = [ValidationError({'field1': 'field error', 'field2': 'err'}), 'simple error']
+        exception_a = ValidationError(error_list)
+        exception_b = ValidationError(list(error_list))
+        self.assertEqual(exception_a, exception_b)
+        self.assertEqual(exception_a.__hash__(), exception_b.__hash__())
+
+        error_list = ['simple error', ValidationError({'field2': 'err', 'field1': 'field error'})]
+        exception_a = ValidationError(error_list)
+        self.assertEqual(exception_a, exception_b)
+        self.assertEqual(exception_a.__hash__(), exception_b.__hash__())
+
+
+        exception_a = ValidationError(
+            [ValidationError({'field1': 'field error', 'field2': 'err'}), 'simple error']
+        )
+        exception_b = ValidationError([ValidationError({'field1': 'field error', 'field2': 'err'}), 'diff error'])
+        self.assertNotEqual(exception_a, exception_b)
+        self.assertNotEqual(exception_a.__hash__(), exception_b.__hash__())
+
+    def test_equality_on_different_message_types(self):
+        # message type is referring to `message` being either a simple str, an `error_list` or `error_dict`
+        exception_str = ValidationError('e1 %(parm1)s', code='my_code', params={'parm1': 'val1'})
+        exception_list = ValidationError(
+            [ValidationError('e1 %(parm1)s', code='my_code', params={'parm1': 'val1'})]
+        )
+        exception_dict = ValidationError(
+            {'field1': ValidationError('e1 %(parm1)s', code='my_code', params={'parm1': 'val1'})}
+        )
+        self.assertNotEqual(exception_str, exception_list)
+        self.assertNotEqual(exception_str.__hash__(), exception_list.__hash__())
```

Comment:
Use `hash()` instead of `__hash__()`, e.g. 
```suggestion
        self.assertNotEqual(hash(exception_str), hash(exception_list))
```